packages/core/src/server/runner/esm.ts

```diff
@@ -6,21 +7,62 @@ import { asModule } from './asModule';
 import { CommonJsRunner } from './cjs';
 import { EsmMode, type RunnerRequirer } from './type';
 
+function findPackageJson(startDir: string) {
+  let dir = path.resolve(startDir);
+  do {
+    const pkgFile = path.join(dir, 'package.json');
+    if (fs.existsSync(pkgFile)) {
```

Comment:
Can we use async fs here

Reply:
No, this method will be executed when static / dynamic import is executed, so a synchronization method is required.

Reply:
Maybe we can have a better way to deal with this, but it will take some extra time 😢